app/javascript/src/NewService/components/FormElements/FormWrapper.jsx

```diff
@@ -0,0 +1,38 @@
+// @flow
+
+import React from 'react'
+import type {FormProps} from 'NewService/types'
+import {CSRFToken} from 'utilities/utils'
+import {HiddenServiceDiscoveryInput} from 'NewService/components/FormElements'
+
+const FormWrapper = (props: FormProps) => {
+  const {id, formActionPath, hasHiddenServiceDiscoveryInput, submitText} = props
+  return (
+    <form
+      className='formtastic service'
+      id={id}
+      action={formActionPath}
+      acceptCharset="UTF-8"
+      method="post"
+    >
+      <input name="utf8" type="hidden" value="✓"/>
+      <CSRFToken />
+      { hasHiddenServiceDiscoveryInput ? <HiddenServiceDiscoveryInput /> : null }
```

Comment:
Here you could get rid of that nasty `null` like : `hasHiddenServiceDiscoveryInput && <HiddenServiceDiscoveryInput />`